### PR TITLE
feat: Flow main layout

### PR DESCRIPTION
**Pull request by caalador**
add `withLayouts` method to RouterConfigurationBuilder
for using Flow serverside view as main layout.
To enable  call `.withLayouts(Flow)` in `routes.tsx`
and add `"layout": true` into the views `ViewConfig`.

### CODE REVIEW

File: packages/ts/file-router/src/types.d.ts [hunk 1]
```diff
@@ -25,6 +25,12 @@ export type ViewConfig = Readonly<{
    */
   route?: string;
 
+  /**
+   * Allows giving marking view as using server side parent layout
```
Comment (taefi) [inline packages/ts/file-router/src/types.d.ts:29]:
```suggestion
   * Set to true to indicate that the view as using server side parent layout
```

Comment (caalador) [reply packages/ts/file-router/src/types.d.ts:29]: That is already possible, but the linking is done on the server so it is not needed as information on the client.

**Review by taefi — CHANGES_REQUESTED** [review]
Generally, looks fine, added a couple of nit picking comments ;)

File: packages/ts/file-router/src/types.d.ts [hunk 2]
```diff
@@ -25,6 +25,12 @@ export type ViewConfig = Readonly<{
    */
   route?: string;
 
+  /**
+   * Set to true to indicate that the view as using server side parent layout
```
Comment (taefi) [inline packages/ts/file-router/src/types.d.ts:29]:
```suggestion
   * Set to true to indicate that the view is using server side parent layout
```

**Review by taefi — CHANGES_REQUESTED** [review]

File: packages/ts/file-router/src/runtime/RouterConfigurationBuilder.ts [hunk 3]
```diff
@@ -143,6 +143,51 @@ export class RouterConfigurationBuilder {
     return this;
   }
 
+  /**
+   * Adds the layoutComponent as the parent layout to views with the flowLayouts ViewConfiguration set.
+   *
+   * @param layoutComponent - layout component to use, usually Flow
+   */
+  withLayouts(layoutComponent: ComponentType): this {
```
[the remark this answers is not in the document]
Comment (caalador) [reply packages/ts/file-router/src/runtime/RouterConfigurationBuilder.ts]: done

**Review by taefi — APPROVED** [review]
LGTM!

**Pull request merged**